inorder-handler.md

```diff
@@ -0,0 +1,73 @@
+# Description of inoder-handler
+
+### Brief about Virtio:  
+virtIO is a virtualization standard for network and disk device drivers. It was chosen to be the main platform for IO virtualization in KVM.
+
+### About virtio block:
+- The virtio-blk device presents a block device to the virtual machine.
+- Each virtio-blk device appears as a disk inside the guest.
+- Virtio-blk is an abstraction layer over devices in a paravirtualized hypervisor.
+- It mainly provides two abstractions:
+  - Request: Contains block request parsing abstraction.
+  - Executor: Contains a block request execution abstraction that is based on std::io::Read and std::io::Write.
+
+### Inorder-handler: 
+
+- This object is used to process the queue of a block device without making any assumptions about the notification mechanism.
+- For time being they used sepecific backend 'StdIoBackend' but in furture they wanted to implement the more generic and flexible one.
+- This queuehandler takes the request and process the request. Return the requests in the same order in which they received.
+```rs
+pub struct InOrderQueueHandler<M: GuestAddressSpace, S: SignalUsedQueue> {
+    pub driver_notify: S,
+    pub queue: Queue<M>,
+    pub disk: StdIoBackend<File>,
+}
+```
+
+- In this process_queue implementation, we follow various steps:
+  - First, we disable the notification event from guest drivers.  
+  - Then, we process the available chain of request through process_chain method.  
+  - If notification is enabled, then we process next set of chain of request.  
```

Comment:
- If notification is enabled, then we process the next set of requests in the chain.